src/main/java/appeng/tile/spatial/SpatialAnchorTileEntity.java

```diff
@@ -0,0 +1,375 @@
+import appeng.api.networking.events.statistics.MENetworkChunkEvent.MENetworkChunkRemoved;
+import appeng.api.networking.ticking.IGridTickable;
+import appeng.api.networking.ticking.TickRateModulation;
+import appeng.api.networking.ticking.TickingRequest;
+import appeng.api.util.AECableType;
+import appeng.api.util.AEColor;
+import appeng.api.util.AEPartLocation;
+import appeng.api.util.DimensionalCoord;
+import appeng.api.util.IConfigManager;
+import appeng.api.util.IConfigurableObject;
+import appeng.client.render.overlay.IOverlayDataSource;
+import appeng.client.render.overlay.OverlayManager;
+import appeng.me.GridAccessException;
+import appeng.services.ChunkLoadingService;
+import appeng.tile.grid.AENetworkTileEntity;
+import appeng.util.ConfigManager;
+import appeng.util.IConfigManagerHost;
+
+public class SpatialAnchorTileEntity extends AENetworkTileEntity
+        implements IGridTickable, IConfigManagerHost, IConfigurableObject, IOverlayDataSource {
+
+    private final ConfigManager manager = new ConfigManager(this);
+    private final Set<ChunkPos> chunks = new HashSet<>();
+    private int powerlessTicks = 0;
+    private boolean initialized = false;
+    private boolean displayOverlay = false;
+
+    public SpatialAnchorTileEntity(TileEntityType<?> tileEntityTypeIn) {
+        super(tileEntityTypeIn);
+        this.getProxy().setFlags(GridFlags.REQUIRE_CHANNEL);
+        this.manager.registerSetting(Settings.OVERLAY_MODE, YesNo.NO);
+    }
+
+    @Override
+    public CompoundNBT write(CompoundNBT data) {
+        super.write(data);
+        this.manager.writeToNBT(data);
+        return data;
+    }
+
+    @Override
+    public void read(BlockState blockState, CompoundNBT data) {
+        super.read(blockState, data);
+        this.manager.readFromNBT(data);
+    }
+
+    @Override
+    protected void writeToStream(PacketBuffer data) throws IOException {
+        super.writeToStream(data);
+        data.writeBoolean(displayOverlay);
+        if (this.displayOverlay) {
+            data.writeLongArray(chunks.stream().mapToLong(ChunkPos::asLong).toArray());
+        }
+    }
+
+    @Override
+    protected boolean readFromStream(PacketBuffer data) throws IOException {
+        boolean ret = super.readFromStream(data);
+        boolean newDisplayOverlay = data.readBoolean();
+        ret = newDisplayOverlay != this.displayOverlay || ret;
+        this.displayOverlay = newDisplayOverlay;
+
+        // Cleanup old data and remove it from the overlay manager as safeguard
+        this.chunks.clear();
+        OverlayManager.getInstance().removeHandlers(this);
+
+        if (this.displayOverlay) {
+            this.chunks.addAll(Arrays.stream(data.readLongArray(null)).boxed().map(c -> new ChunkPos(c))
+                    .collect(Collectors.toSet()));
+            // Register it again to render the overlay
+            OverlayManager.getInstance().showArea(this);
```

Comment:
I am not the biggest fan doing this from readStream, but oh well, if it works...

Reply:
p.s.: how do you decide to show the overlay when the chunkloader is not in view? 
and: how do you remove the overlay when it goes out of reach

Reply:
It's just an entry into a map. But I am also not real fan of. However there is no ticking for our tiles client side, so it has to react to something.

Reply:
In terms of distance, not at all. Minecraft itself stops rendering it after like 600 blocks and the source itself is there to not unload itself. While it is outside the normal chunk render distance, it is still easily visible and could still act as guide for not that far away sections. And with the way the anchor works, it also only reports the chunk position of the current world.